internal/dataplane/translator/subtranslator/ingress.go

```diff
@@ -179,18 +242,75 @@ type ingressTranslationMeta struct {
 	ingressUID       string
 	ingressHost      string
 	ingressTags      []*string
-	serviceName      string
-	servicePort      kongstate.PortDef
+	backend          ingressTranslationMetaBackend
 	paths            []netv1.HTTPIngressPath
 	addRegexPrefixFn addRegexPrefixFn
 }
 
+type ingressPathBackendType string
+
+const (
+	ingressPathBackendTypeKongServiceFacade ingressPathBackendType = "KongServiceFacade"
+)
+
+type ingressTranslationMetaBackend struct {
+	// backendType is the type of backend. If left empty, it's assumed to be a Kubernetes Service.
+	backendType ingressPathBackendType
+
+	// name is the Kubernetes object name of the backend.
+	name string
+
+	// port is the port of the backend.
+	port kongstate.PortDef
+}
+
+// intoKongRouteName constructs a Kong Route name for the ingressTranslationMeta object.
+func (b ingressTranslationMetaBackend) intoKongRouteName(ingress k8stypes.NamespacedName, host string) string {
+	// For KongServiceFacade backends, Kong Routes are created separately for the following combination
+	// `<ingress-namespace>.<ingress-name>.<host>.<service-facade-name>.svc.facade`.
+
+	// Note: in the case of KongServiceFacade, we don't use the port in the Kong Route name
+	// because KongServiceFacade may only specify one port (unlike Kubernetes Service) so it's not necessary.
+	if b.backendType == ingressPathBackendTypeKongServiceFacade {
+		return fmt.Sprintf("%s.%s.%s.%s.svc.facade", ingress.Namespace, ingress.Name, host, b.name)
+	}
+
+	// Otherwise, we assume it's a Kubernetes Service and create Kong Routes for the following combination
+	// `<ingress-namespace>.<ingress-name>.<host>.<service-name>.<service-port>`.
+	return fmt.Sprintf("%s.%s.%s.%s.%s", ingress.Namespace, ingress.Name, b.name, host, b.port.CanonicalString())
+}
+
 func (m *ingressTranslationMeta) translateIntoKongStateService(kongServiceName string, portDef kongstate.PortDef) kongstate.Service {
+	if m.backend.backendType == ingressPathBackendTypeKongServiceFacade {
+		return kongstate.Service{
+			Namespace: m.parentIngress.GetNamespace(),
+			Service: kong.Service{
+				Name:           kong.String(kongServiceName),
+				Host:           kong.String(fmt.Sprintf("%s.%s.svc.facade", m.parentIngress.GetNamespace(), m.backend.name)),
```

Comment:
`parentIngress` will be changed to a service in https://github.com/Kong/kubernetes-ingress-controller/issues/5248, right?

Reply:
Not necessarily `ingressTranslationMeta.parentIngress`, but `kongstate.Service.Parent` should be `Service` or `KongServiceFacade`.

Reply:
I see, you're right